include/spdlog/details/async_log_helper.h

```diff
@@ -261,7 +271,7 @@ inline void spdlog::details::async_log_helper::set_formatter(formatter_ptr msg_f
 
 
 // sleep,yield or return immediatly using the time passed since last message as a hint
-inline void spdlog::details::async_log_helper::sleep_or_yield(const clock::time_point& last_op_time)
+inline std::chrono::nanoseconds spdlog::details::async_log_helper::sleep_or_yield(const clock::time_point& last_op_time)
```

Comment:
Then we need to call ::now() again (twice) which is relatively expensive. I can make sleep_or_yield() accepting time_since_op not last_op_time though.